chroma-manager/tests/unit/chroma_core/plugins/test_block_devices.py

```diff
@@ -159,19 +160,125 @@
 #        self._start_session_with_data(host1, "NoDevices.json")
 #        self.assertEqual(Volume.objects.count(), 0)
 
+class TestBase(unittest.TestCase):
+    test_host_fqdn = 'vm5.foo.com'
+
+    def setUp(self):
+        super(TestBase, self).setUp()
+        self.test_root = os.path.join(os.path.dirname(__file__), "fixtures")
+        self.addCleanup(patch.stopall)
+
+    def load(self, filename):
+        return open(os.path.join(self.test_root, filename)).read()
+
+    def check(self, skip_keys, expect, result, x):
+        from toolz import pipe
+        from toolz.curried import map as cmap, filter as cfilter
 
-class TestBlockDevices(unittest.TestCase):
+        def cmpval(key):
+            expected = expect[x][key]
+            actual = result[x][key]
+            if type(expected) is dict:
+                self.check(skip_keys, expect[x], result[x], key)
+            else:
+                self.assertEqual(actual, expected,
+                                 "item {} ({}) in {} does not match expected ({})".format(key,
+                                                                                          actual,
+                                                                                          x,
+                                                                                          expected))
+
+        pipe(expect[x].keys(),
+             cfilter(lambda y: y not in skip_keys),
+             cmap(cmpval),
+             list)
+
+
+class TestFormattedBlockDevices(TestBase):
+    """ Verify aggregator output parsed through block_devices matches expected agent output """
+
+    def setUp(self):
+        super(TestFormattedBlockDevices, self).setUp()
+
+        self.test_root = os.path.join(os.path.dirname(__file__), "fixtures")
+        self.fixture = json.loads(self.load(u'device_aggregator_formatted_ldiskfs.text'))
+        self.block_devices = self.get_patched_block_devices(dict(self.fixture))
+        self.expected = json.loads(self.load(u'agent_plugin_formatted_ldiskfs.json'))['result']['linux']
+
+        self.addCleanup(patch.stopall)
+
+    def get_patched_block_devices(self, fixture):
+        with patch('chroma_core.plugins.block_devices.aggregator_get', return_value=fixture):
+            return get_block_devices(self.test_host_fqdn)
+
+    def test_block_device_nodes_parsing(self):
+        p = re.compile('\d+:\d+$')
+        print 'Omitted devices:'
+        print (set(self.expected['devs'].keys()) - set([mm for mm in self.expected['devs'].keys() if p.match(mm)]))
+        ccheck = curry(self.check, [], self.expected['devs'], self.block_devices['devs'])
+
+        map(
```

Comment:
This works since you're doing an assert but another option is to map to a boolean and then use a function that checks that every element is True. The reason I mention this is that it feels odd that map isn't returning anything.

Reply:
can look at revising this later, this code may well change shortly as block_devices disappears